src/OVAL/oval_schema_version.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2015 Red Hat Inc., Durham, North Carolina.
+ * All Rights Reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Authors:
+ *      "Jan Černý" <jcerny@redhat.com>
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include "common/util.h"
+#include "common/debug_priv.h"
+#include "public/oval_schema_version.h"
+
+#if defined USE_REGEX_PCRE
+#include <pcre.h>
+#elif defined USE_REGEX_POSIX
+#include <regex.h>
+#endif
+
+enum {
+	OVAL_SCHEMA_VERSION_CORE_MAJOR,
+	OVAL_SCHEMA_VERSION_CORE_MINOR,
+	OVAL_SCHEMA_VERSION_CORE_UPDATE,
+	OVAL_SCHEMA_VERSION_PLATFORM_MAJOR,
+	OVAL_SCHEMA_VERSION_PLATFORM_MINOR,
+	OVAL_SCHEMA_VERSION_PLATFORM_UPDATE
+};
+
+static int _parse_int(const char *substring, size_t substring_length)
+{
+	/* Pay attention that substring_length != strlen(substring) */
+	char buffer[substring_length + 1]; // +1 for a zero byte
+	strncpy(buffer, substring, substring_length);
+	buffer[substring_length] = '\0';
+	return atoi(buffer);
+}
+
+oval_schema_version_t oval_schema_version_from_cstr(const char *ver_str)
+{
+	oval_schema_version_t version;
+	memset(&version, 0, sizeof(oval_schema_version_t));
+	if (ver_str == NULL) {
+		return version;
+	}
+#if defined USE_REGEX_PCRE
```

Comment:
Isn't there any interface providing some abstraction to work with regex? This code contains duplicity...